lib/checkbool.cpp

```diff
@@ -429,8 +430,8 @@ void CheckBool::pointerArithBoolError(const Token *tok)
     reportError(tok,
                 Severity::error,
                 "pointerArithBool",
-                "Converting pointer arithmetic result to bool. The bool is always true unless there is undefined behaviour.\n"
-                "Converting pointer arithmetic result to bool. The boolean result is always true unless there is pointer arithmetic overflow, and overflow is undefined behaviour. Probably a dereference is forgotten.", CWE571, false);
+                "Converting pointer arithmetic result to bool. The bool is always true unless there is undefined behavior.\n"
+                "Converting pointer arithmetic result to bool. The boolean result is always true unless there is pointer arithmetic overflow, and overflow is undefined behavior. Probably a dereference is forgotten.", CWE571, false);
```

Comment:
I would prefer to keep the message as it is. Anyway, if the message is changed, the test suite needs to be changed, too, otherwise there are test failures.